mobile-app/lib/features/components/transactions_list.dart

```diff
@@ -82,7 +63,7 @@ class RecentTransactionsList extends StatelessWidget {
                     return TransactionListItem(
                       key: ValueKey(transaction.id),
                       transaction: transaction,
-                      role: _getTransactionRole(transaction),
+                      role: TransactionUtils.getTransactionRole(transaction, accountIds),
```

Comment:
If this is the only reason we pass account ids around everywhere, I would suggest a different architecture to check if "this account is one of my accounts". 

We can check with settings for example, which would be equivalent to passing around this setting, but ideally it should be connected to account provider so we can dynamically update everything automatically whenever accounts list changes.